--- cleantalk.class.php	2024-08-06 20:01:35
+++ cleantalk_patch.class.php	2024-08-06 20:42:42
@@ -745,7 +745,7 @@
      * @return array
      */
     public function get_servers_ip($host) {
-        $response = null;
+        $response = array();
         if (!isset($host))
             return $response;
 
@@ -780,7 +780,7 @@
         } else {
             // $i - to resolve collisions with localhost
             $i = 0;
-            $r_temp = null;
+            $r_temp = array();
             $fast_server_found = false;
             foreach ($response as $server) {
                 
